awkward-cpp/CMakeLists.txt

```diff
@@ -25,15 +25,11 @@ Please check https://github.com/scikit-hep/awkward#installation-for-developers t
 endif()
 
 # Setup the RPATH for built libraries
-if(CMAKE_SYSTEM_NAME STREQUAL Emscripten)
-  message(STATUS "Using Emscripten")
-  set(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
-  set(CMAKE_BUILD_RPATH_USE_ORIGIN TRUE)
-elseif(APPLE)
-  set(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
+set(CMAKE_BUILD_RPATH_USE_ORIGIN TRUE)
+if(APPLE)
   set(CMAKE_INSTALL_RPATH "@loader_path")
 else()
-  set(CMAKE_BUILD_RPATH_USE_ORIGIN TRUE)
+  set(CMAKE_INSTALL_RPATH "\$ORIGIN")
```

Comment:
Previously I'd been under the impression that `CMAKE_BUILD_RPATH_USE_ORIGIN` _sets_ the `RPATH` to `$ORIGIN`. But, it appears that instead this variable just makes it possible to use relative paths with `$ORIGIN`. As such, we need to set `CMAKE_INSTALL_RPATH` in addition to enabling` $ORIGIN` on platforms for which it is supported.

Reply:
N.B. This was caught by auditwheel, which complained that `_ext`'s new link to `libawkward` couldn't be resolved. In the past, `_ext` was statically linked against awkward, so we didn't have to resolve the `RPATH`.